src/jit/hwintrinsicxarch.cpp

```diff
@@ -445,16 +570,40 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
     {
         return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
     }
-    else if (category == HW_Category_IMM)
+    // Avoid checking stacktop for 0-op intrinsics
+    if (sig->numArgs > 0 && isImmHWIntrinsic(intrinsic, impStackTop().val))
```

Comment:
Why `if` instead of `else if`?

Reply:
From here, each `if` is not exclusive to others. For example, an intrinsic could be true at `if (sig->numArgs > 0 && isImmHWIntrinsic(intrinsic, impStackTop().val))` and `if ((flags & (HW_Flag_OneTypeGeneric | HW_Flag_TwoTypeGeneric)) != 0)`.